primitives/Cargo.toml

```diff
@@ -0,0 +1,60 @@
+[package]
+name = "bitcoin-primitives"
+# Arbitrary high version number so as to not clash with the original
+# `bitcoin-primitives` crate that we replaced `v0.1.16-alpha`.
+version = "0.100.0"
+authors = ["Andrew Poelstra <apoelstra@wpsoftware.net>"]
+license = "CC0-1.0"
+repository = "https://github.com/rust-bitcoin/rust-bitcoin"
+description = "Primitive types used by the rust-bitcoin eccosystem"
+categories = ["cryptography::cryptocurrencies"]
+keywords = ["bitcoin", "types"]
+readme = "README.md"
+edition = "2021"
+rust-version = "1.56.1"
+exclude = ["tests", "contrib"]
+
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
+
+[features]
+default = ["std"]
+std = ["hashes/std", "hex/std", "internals/std", "io/std", "units/std"]
+
+# Note, optional dependencies `base58`, `bech32`, and `bitcoinconsensus` do not have a serde feature.
+serde = ["actual-serde", "hashes/serde", "hex/serde", "internals/serde", "units/serde"]
+
+# Features for optional dependencies that have a `std` feature that needs enabling for std builds.
+bitcoinconsensus-std = ["bitcoinconsensus/std", "std"]
+base58-std = ["base58/std", "std"]
+bech32-std = ["bech32/std", "std"]
+
+# Enables transient dependency on the `rand` crate.
+rand-std = ["secp256k1/rand-std", "std"]
+
+[dependencies]
+hashes = { package = "bitcoin_hashes", version = "0.14.0", default-features = false, features = ["alloc", "io"] }
+hex = { package = "hex-conservative", version = "0.2.0", default-features = false, features = ["alloc"] }
+internals = { package = "bitcoin-internals", version = "0.3.0", features = ["alloc"] }
+io = { package = "bitcoin-io", version = "0.1.1", default-features = false, features = ["alloc"] }
+units = { package = "bitcoin-units", version = "0.1.0", default-features = false, features = ["alloc"] }
+
+# Only use these features for no-std builds, otherwise use the appropriate feature eg, `base58-std`.
+base58 = { package = "base58ck", version = "0.1.0", default-features = false, optional = true }
+bech32 = { version = "0.11.0", default-features = false, features = ["alloc"], optional = true }
+bitcoinconsensus = { version = "0.106.0+26", default-features = false, optional = true }
+
+# Enables implementations of `ArbitraryOrd` on some structs.
+ordered = { version = "0.2.0", optional = true }
+
+# Enables cryptography by way of `rust-secp256k1`.
+secp256k1 = { version = "0.29.0", default-features = false, features = ["hashes"], optional = true }
+
+# Do NOT use this as a feature! Use the `serde` feature instead.
+actual-serde = { package = "serde", version = "1.0.103", default-features = false, features = [ "derive", "alloc" ], optional = true }
```

Comment:
For something that's supposed to be primitives AKA entities under Clean Architecture terminology, this has a shitton of dependencies. If I were to be entirely strict this should be the dependency of almost everything else instead, however this is Rust and we need to break the rules in some cases (`serde`) and we need to do a little dependency dance in some other cases. **Still assuming our primary goal is to get 1.0 out ASAP.**

My suggestion for dependencies:

* `hashes` - obviously, non-public dependency. This implies we either must NOT implement any traits from `hashes` or stabilize hashes soon; I think we can do the former which puts us at ease, allows stabilization sooner and actually might help with split hash trait thing. We're able to do this at this point, I think.
`units`,
* `units` - public dependency, obviously, easy to stabilize, we use `Amount` in `TxOut`.
* `internals` - private
* `hex` - private
* `ordered` - needs to be stabilized, otherwise we'll have to remove it; required because of external trait
* `secp256k1` - private (for now), though I'm not sure about `XOnlyPublicKey`. Required for keys. (Though maybe we don't strictly need them - we could impl `Script` methods as extension trait in `bitcoin`. OTOH, keys seem to be an important part of the ecosystem.)
* `serde` - already stable, required because we're implementing its traits

Address-related deps should go to address, consensus verification can for now live in `bitcoin`, `io` in `bitcoin-consensus-encoding`.